src/main/java/no/nav/familie/integrasjoner/client/rest/DokarkivRestClient.kt

```diff
@@ -0,0 +1,68 @@
+package no.nav.familie.integrasjoner.client.rest
+
+import no.nav.familie.http.util.UriUtil
+import no.nav.familie.integrasjoner.dokarkiv.client.KanIkkeFerdigstilleJournalpostException
+import no.nav.familie.integrasjoner.dokarkiv.client.domene.FerdigstillJournalPost
+import no.nav.familie.integrasjoner.dokarkiv.client.domene.OpprettJournalpostRequest
+import no.nav.familie.integrasjoner.dokarkiv.client.domene.OpprettJournalpostResponse
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestClientResponseException
+import org.springframework.web.client.RestOperations
+import java.net.URI
+
+@Component
+class DokarkivRestClient(@Value("\${DOKARKIV_V1_URL}") private val dokarkivUrl: URI,
+                         @Qualifier("sts") private val restOperations: RestOperations)
+    : AbstractPingableRestClient(restOperations, "dokarkiv.opprett") {
+
+    override val pingUri: URI = UriUtil.uri(dokarkivUrl, PATH_PING)
+
+    private val ferdigstillJournalPostClient = FerdigstillJourrnalPostClient(restOperations, dokarkivUrl)
+
+    fun lagJournalpostUri(ferdigstill: Boolean): URI =
+            UriUtil.uri(dokarkivUrl, PATH_JOURNALPOST, String.format(QUERY_FERDIGSTILL, ferdigstill))
+
+    fun lagJournalpost(jp: OpprettJournalpostRequest,
+                       ferdigstill: Boolean,
+                       personIdent: String?): OpprettJournalpostResponse? {
+        val uri = lagJournalpostUri(ferdigstill)
+        return postForEntity(uri, jp)
+    }
+
+    fun ferdigstillJournalpost(journalpostId: String, journalførendeEnhet: String) {
+        ferdigstillJournalPostClient.ferdigstillJournalpost(journalpostId, journalførendeEnhet)
+    }
+
+    /**
+     * Privat klasse for å gi egne metrics for ferdigstilling av jpurnalpost.
+     *
+     */
+    private class FerdigstillJourrnalPostClient(restOperations: RestOperations, private val dokarkivUrl: URI)
+        : AbstractRestClient(restOperations, "dokarkiv.ferdigstill") {
+
+        private fun ferdigstillJournalpostUri(journalpostId: String): URI {
+            return UriUtil.uri(dokarkivUrl, String.format(PATH_FERDIGSTILL_JOURNALPOST, journalpostId))
+        }
+
+        fun ferdigstillJournalpost(journalpostId: String, journalførendeEnhet: String) {
+            val uri = ferdigstillJournalpostUri(journalpostId)
+            try {
+                patchForEntity<Any>(uri, FerdigstillJournalPost(journalførendeEnhet))
+            } catch (e: RuntimeException) {
+                if (e.cause is RestClientResponseException && (e.cause as RestClientResponseException).rawStatusCode == 400) {
```

Comment:
Kan skrives som 
if (e.cause is HttpClientErrorException.BadRequest)

Reply:
Fikset